### DISCUSSION TITLE
pandas 101

### DISCUSSION THREAD

**Discussion by TinyHust** [TinyHust/notes#438] [Vault]
**REFs:**
- https://github.com/mediumhust/100_bai_tap_pandas
- https://github.com/mediumhust/complete-pandas-tutorial

**Comment by mediumhust** [comment]
flatten a hierarchical index in columns
```python
df_tmp = df_receipt.groupby('customer_id').agg({'sales_ymd':['max','min']}).reset_index()

df_tmp.columns = [" ".join(pair).strip() for pair in df_tmp.columns]

df_tmp.query('`sales_ymd max` != `sales_ymd min`').head(10)
```

**Comment by mediumhust** [comment]
cut & qcut
- https://www.geeksforgeeks.org/how-to-use-pandas-cut-and-qcut/
- https://www.geeksforgeeks.org/pandas-cut-method-in-python/
